SConstruct

```diff
@@ -1027,6 +1027,16 @@ def get_expression_value(includes, expression, defines=()):
               '}\n'))
     return '\n'.join(s)
 
+# Check that NaN is treated correctly
+nan_check_source = get_expression_value(["<cmath>"], 'std::isnan(NAN + argc)')
+retcode, nan_works = conf.TryRun(nan_check_source, ".cpp")
+if nan_works.strip() != "1":
+    config_error(
+        "Cantera requires a working implementation of 'std::isnan'.\n"
+        "If you have specified '-ffast-math' or equivalent as an optimization option\n"
+        "either remove this option or add the '-fno-finite-math-only option"
```

Comment:
Missing `'`. Also, `-fno-finite-math-only` does not exist on all compilers.

Reply:
Do you know which compilers so the comment can include that info?

Reply:
For the intel compilers, you have to disable fast-math altogether. The option `-fp-model fast` includes this optimization and does not work (see default options for `icc`/`icx` in `SConstruct`) ... see [documentation](https://www.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/compiler-options/compiler-option-details/floating-point-options/fp-model-fp.html)

Reply:
> Regarding `self.assertNear` vs. `self.assertEqual`: equivalent changes probably should be applied across the test suite whenever floats are compared? (which would be a _lot_ of churn, so I'm not exactly advocating for this here.)

While it's not obvious from the diff alone, the instances I changed are more than just the test cases that failed -- these are the other instances in these files where a calculated value was being compared (because we go back and forth between Ea/R and E).

Reply:
While `icx' doesn't list this flag in its man page, I tested this and it does work. I guess it's implemented for the sake of compatibility with GCC's options.

Reply:
That's great to know. I guess google was not my friend here :rofl: